lib/ansible/modules/cloud/amazon/ec2_ami_copy.py

```diff
@@ -237,23 +204,21 @@ def main():
 
     if not HAS_BOTO:
         module.fail_json(msg='boto required for this module')
+    # TODO: Check botocore version
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
 
-    try:
-        ec2 = ec2_connect(module)
-    except boto.exception.NoAuthHandlerFound as e:
-        module.fail_json(msg=str(e))
-
-    try:
-        region, ec2_url, boto_params = get_aws_connection_info(module)
-    except boto.exception.NoAuthHandlerFound as e:
-        module.fail_json(msg=str(e))
+    if HAS_BOTO3:
 
-    if not region:
-        module.fail_json(msg="region must be specified")
+        try:
+            ec2 = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url,
+                             **aws_connect_params)
+        except NoRegionError:
+            module.fail_json(msg='AWS Region is required')
+    else:
+        module.fail_json(msg='boto3 required for this module')
 
-    copy_image(module, ec2)
+    copy_image(ec2, module)
```

Comment:
code changes look good ... some look unneeded like the order of arguments above ... but not show stopper.